internal/telemetry/event.go

```diff
@@ -0,0 +1,66 @@
+package telemetry
+
+import (
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Event is a telemetry event
+type Event struct {
+	id          string
+	eventType   EventType
+	userID      string
+	time        time.Time
+	executionID string
+	command     string
+	data        map[DataKey]interface{}
```

Comment:
for ease, this could probably just become a `[]EventData` until we figure out how we want to transform that data into a Segment construct